src/foundry/foundry.js/pixi/containers/canvasLayers/placeablesLayers/wallsLayer.d.ts

```diff
@@ -320,7 +320,7 @@ declare global {
     interface LayerOptions extends PlaceablesLayer.LayerOptions<'Wall'> {
       name: 'walls';
       controllableObjects: true;
-      objectClass: ConstructorOf<Wall>;
+      objectClass: ConfiguredObjectClassForName<'Wall'>;
```

Comment:
Just one hint without looking at everything: `WallsLayer.layerOptions()` returns `objectClass: Wall`. When changing the `LayerOptions` here, we might adjust the return type of `WallsLayer.layerOptions()`.

Reply:
To be honest, I think that's a bug in foundry. `Wall` shouldn't be hardcoded there.

Reply:
Hmm, it looks like LayerOptions.objectClass is actually the Document class for most layers.

![image](https://user-images.githubusercontent.com/182554/130667739-4d5852d8-9f6b-460f-ae06-8dd4763174d4.png)

I wonder if this is a bug in foundry or intended behavior?

Reply:
It is apparently a bug. https://discord.com/channels/170995199584108546/848968683623022632/879805304110456963